src/aosm/azext_aosm/generate_nsd/nf_ret.py

```diff
@@ -0,0 +1,165 @@
+# --------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT
+# License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------
+"""Handles the creation of a resource element template for a network function."""
+
+import json
+
+from typing import Dict, Any, List, Union
+from knack.log import get_logger
+
+from azext_aosm._configuration import NFDRETConfiguration
+from azext_aosm.util.constants import CNF, VNF
+from azext_aosm.util.management_clients import ApiClients
+from azext_aosm.vendored_sdks.models import NetworkFunctionDefinitionVersion, NFVIType
+
+
+logger = get_logger(__name__)
+
+
+class nfRET:
+    """
+    Represents a single network function resource element template withing an NSD.
+    """
+
+    def __init__(
+        self, api_clients: ApiClients, config: NFDRETConfiguration, cg_schema_name: str
+    ) -> None:
+        self.config = config
+        self.cg_schema_name = cg_schema_name
+        nfdv = self._get_nfdv(config, api_clients)
+        print(
+            f"Finding the deploy parameters for {self.config.name}:{self.config.version}"
+        )
+
+        if not nfdv.deploy_parameters:
+            raise NotImplementedError(
+                f"NFDV {self.config.name} has no deploy parameters, cannot generate NSD."
+            )
+        self.deploy_parameters: Dict[str, Any] = json.loads(nfdv.deploy_parameters)
+
+        self.nf_type = self.config.name.replace("-", "_")
+        self.nfdv_parameter_name = f"{self.nf_type}_nfd_version"
+        self.config_mapping_filename = f"{self.config.name}_config_mapping.json"
+
+    def _get_nfdv(
+        self, config: NFDRETConfiguration, api_clients
+    ) -> NetworkFunctionDefinitionVersion:
+        """Get the existing NFDV resource object."""
+        print(
+            "Reading existing NFDV resource object "
+            f"{config.version} from group {config.name}"
+        )
+        nfdv_object = api_clients.aosm_client.network_function_definition_versions.get(
+            resource_group_name=config.publisher_resource_group,
+            publisher_name=config.publisher,
+            network_function_definition_group_name=config.name,
+            network_function_definition_version_name=config.version,
+        )
+        return nfdv_object
+
+    @property
+    def config_mappings(self) -> Dict[str, Any]:
+        """
+        Return the contents of the config mapping file for this RET.
+        """
+        nf = self.config.name
+
+        logger.debug("Create %s", self.config_mapping_filename)
+
+        deployment_parameters: Union[str, List[str]] = f"{{configurationparameters('{self.cg_schema_name}').{nf}.deploymentParameters}}"
```

Comment:
Again, I appreciate that you have only moved this, but I'm finding this old code tricky to follow.  Think this function could do with some examples of WTF is going on.